spec/tasks/assets_tasks_spec.rb

```diff
@@ -0,0 +1,40 @@
+require "rails_helper"
+
+RSpec.describe "asset tasks" do
+  Rails.application.load_tasks
+
+  describe "update_topical_event_legacy_url_paths" do
+    let(:task) { Rake::Task["assets:update_topical_event_legacy_url_paths"] }
+
+    def legacy_url(asset_name, model_name)
+      "/government/uploads/system/uploads/#{model_name}/file/1234/#{asset_name}.jpeg"
+    end
+
+    def assert_assets_the_same_except_legacy_url_path(actual_assets, expected_assets)
+      ignored_keys = %w[legacy_url_path updated_at last_modified created_at] # annoyingly there's a really small difference in created at which is I think due to mongo truncating
+      assets_except_ignored_keys = ->(assets) { assets.map { |asset| asset.attributes.except(*ignored_keys) } }
+      expect(assets_except_ignored_keys.call(actual_assets)).to eq assets_except_ignored_keys.call(expected_assets)
+    end
+
+    it "updates any assets with legacy urls matching /classification_featuring_image_data to /topical_event_image_data" do
+      classification_featuring_urls = (1..5).map { |i| legacy_url("asset-#{i}", "classification_featuring_image_data") }
+      classification_featuring_assets = classification_featuring_urls.map { |url| FactoryBot.create(:whitehall_asset, legacy_url_path: url) }
+      non_matching_url = legacy_url("asset-10", "not_a_classification_featuring_image_data")
+      non_matching_asset = FactoryBot.create(:whitehall_asset, legacy_url_path: non_matching_url)
+
+      task.invoke
+
+      ## There should now be 5 assets with an updated path, whose urls match and which are equivalent to the assets before they were renamed
+      topical_event_featuring_assets = WhitehallAsset.where(legacy_url_path: /\/government\/uploads\/system\/uploads\/topical_event_featuring_image_data/)
+      expected_urls = (1..5).map { |i| legacy_url("asset-#{i}", "topical_event_featuring_image_data") }
+      expect(topical_event_featuring_assets.pluck(:legacy_url_path)).to eq expected_urls
+      assert_assets_the_same_except_legacy_url_path(topical_event_featuring_assets.to_a, classification_featuring_assets)
+
+      ## We shouldn't have any assets left in the database with the classification featuring image data path
```

Comment:
I feel like having to write these comments is probably an indication that you're performing multiple tests in one `it` block. 

I imagine it may be distinct tests like:

```
it "replaces the legacy_url_path of topical event featurings" do
  # create x matching items
  task.invoke
  # expect count of old to be 0 
  # expect count of new to be x
end

it "changes only the legacy_url_path" do
# create one item
task.invoke
# compare attributes
end
```

I'm not sure you need one for that it doesn't change other items, I think that'd be covered implicitly in the first test